amf-validation/shared/src/main/scala/amf/plugins/features/validation/AMFValidator.scala

```diff
@@ -0,0 +1,162 @@
+package amf.plugins.features.validation
+
+import amf.client.execution.BaseExecutionEnvironment
+import amf.client.plugins.{AMFDocumentPlugin, AMFValidationPlugin}
+import amf.core.annotations.SourceVendor
+import amf.core.benchmark.ExecutionLog
+import amf.core.errorhandling.AmfStaticReportBuilder
+import amf.core.model.document.{BaseUnit, Document, Fragment, Module}
+import amf.core.rdf.RdfModel
+import amf.core.registries.AMFPluginsRegistry
+import amf.core.remote.{Oas30, Raml08, Vendor}
+import amf.core.services.RuntimeValidator.{CustomShaclFunctions, validatorOption}
+import amf.core.services.{RuntimeValidator, ValidationOptions => LegacyValidationOptions}
+import amf.core.unsafe.PlatformSecrets
+import amf.core.validation.core.{ValidationProfile, ValidationReport, ValidationSpecification}
+import amf.core.validation.{AMFValidationReport, EffectiveValidations}
+import amf.internal.environment.Environment
+import amf.plugins.features.validation.emitters.{JSLibraryEmitter, ValidationJSONLDEmitter}
+import amf._
+import amf.client.remod.amfcore.plugins.validate.{AMFValidatePlugin, ValidationOptions}
+
+import scala.concurrent.Future.successful
+import scala.concurrent.{ExecutionContext, Future}
+
+protected[amf] trait AMFValidator extends RuntimeValidator with PlatformSecrets {
+  protected var customValidationProfiles: Map[String, () => ValidationProfile]  = Map.empty
+  protected var customValidationProfilesPlugins: Map[String, Seq[AMFValidatePlugin]] = Map.empty
+
+  // All the profiles are collected here, plugins can generate their own profiles
+  protected def profiles: Map[String, () => ValidationProfile] =
+    AMFPluginsRegistry.documentPlugins.foldLeft(Map[String, () => ValidationProfile]()) {
+      case (acc, domainPlugin: AMFValidationPlugin) => acc ++ domainPlugin.domainValidationProfiles(platform)
+      case (acc, _)                                 => acc
+    } ++ customValidationProfiles
+
+  // Mapping from profile to domain plugin
+  protected def profilesPlugins: Map[String, Seq[AMFValidatePlugin]] =
+    AMFPluginsRegistry.documentPlugins.foldLeft(Map[String, Seq[AMFValidatePlugin]]()) {
+      case (acc, domainPlugin: AMFValidationPlugin) =>
+        val toPut = domainPlugin.domainValidationProfiles(platform).keys.foldLeft(Map[String, Seq[AMFValidatePlugin]]()) {
+          case (accProfiles, profileName) =>
+            accProfiles.updated(profileName, domainPlugin.getRemodValidatePlugins())
```

Comment:
noticed there is no usage of the BaseEnvironment to obtain validation plugins, could all registered validation plugins be filtered at this point given certain profile? maybe make use of the applies method.

Reply:
Yes, it can be done. The thing is that there is a path where custom validation uses the `baseProfile` for validation. It isn't clear how we would do the `applies` method of each plugin yet. That is why it isn't used.

Reply:
There is a plan to make validation plugins into "factories" that create the already loaded validators. If this pattern emerges in most plugins then the applies makes no sense in these plugins